trieste/acquisition/function.py

```diff
@@ -199,6 +199,7 @@ def prepare_acquisition_function(
             raise ValueError("Dataset must be populated.")
 
         query_points = self._search_space.sample(num_samples=self._grid_size)
+        tf.debugging.assert_same_float_dtype([dataset.query_points, query_points])
         query_points = tf.concat([dataset.query_points, query_points], 0)
         fmean, fvar = model.predict(query_points)
         fsd = tf.math.sqrt(fvar)
```

Comment:
Nice check. Although wont we get the same error from the line below anyway?

Reply:
The error message in the line below is a tiny bit more cryptic in that it doesn't make it obvious that either type is ok as long as they're consistent.

Reply:
Sure. Maybe it would be better to just force the samples to be the same dtype as the query points?

You might be able to specify their type when you sample them?

Reply:
I think an error is probably better than silently changing the sample dtype, but not too fussed either way. If you're ok, I might check this in as is.

Reply:
Yep go for it!